anchor/src/setup.rs

```diff
@@ -0,0 +1,243 @@
+    }
+
+    fn apt_get<P: AsRef<Path>>(bin: P) -> Result<(), failure::Error> {
+        let status = Command::new("sudo")
+            .arg("apt-get")
+            .arg("install")
+            .arg(bin.as_ref())
+            .arg("-y")
+            .status()?;
+
+        if !status.success() {
+            bail!("Running apt-get failed!");
+        }
+
+        Ok(())
+    }
+
+    fn pacman<P: AsRef<Path>>(bin: P) -> Result<(), failure::Error> {
+        let status = Command::new("sudo")
+            .arg("pacman")
+            .arg("-Syu")
+            .arg(bin.as_ref())
+            .arg("--noconfirm")
+            .status()?;
+
+        if !status.success() {
+            bail!("Running pacman failed!");
+        }
+
+        Ok(())
+    }
+
+    fn yum<P: AsRef<Path>>(bin: P) -> Result<(), failure::Error> {
+        let status = Command::new("sudo")
+            .arg("yum")
+            .arg("install")
+            .arg(bin.as_ref())
+            .arg("-y")
+            .status()?;
+
+        if !status.success() {
+            bail!("Running yum failed!");
+        }
+
+        Ok(())
+    }
+
+    fn brew<P: AsRef<Path>>(bin: P) -> Result<(), failure::Error> {
+        let status = Command::new("brew")
+            .arg("install")
+            .arg(bin.as_ref())
+            .status()?;
+
+        if !status.success() {
+            bail!("Running brew failed!");
+        }
+
+        Ok(())
+    }
+
+    fn install_binary<P: AsRef<Path>>(&self, bin: P) -> Result<(), failure::Error> {
+        match self {
+            PackageManager::AptGet => Self::apt_get(bin),
+            PackageManager::Pacman => Self::pacman(bin),
+            PackageManager::Yum => Self::yum(bin),
+            PackageManager::Brew => Self::brew(bin),
+        }
+    }
+
+    fn install_cmake(&self) -> Result<(), failure::Error> {
+        if is_bin_on_path("cmake")? {
+            return Ok(());
+        }
+
+        println!("Installing cmake....");
+        self.install_binary("cmake")
+    }
+
+    fn install_go(&self) -> Result<(), failure::Error> {
+        if is_bin_on_path("go")? {
+            return Ok(());
+        }
+
+        println!("Installing go....");
+        match self {
+            PackageManager::AptGet => Self::apt_get("golang"),
+            PackageManager::Pacman => Self::pacman("go"),
+            PackageManager::Yum => Self::yum("golang"),
+            PackageManager::Brew => Self::brew("go"),
+        }
+    }
+
+    fn install_unzip(&self) -> Result<(), failure::Error> {
+        if is_bin_on_path("unzip")? {
+            return Ok(());
+        }
+
+        println!("Installing unzip....");
+        self.install_binary("unzip")
+    }
+
+    fn install_protobuf(&self) -> Result<(), failure::Error> {
+        if is_bin_on_path("protoc")? {
+            return Ok(());
+        }
+
+        println!("Installing protobuf....");
+        match self {
+            PackageManager::Brew => Self::brew("protobuf"),
+            _ => {
```

Comment:
spell out the other possibilities? so that if we add support for e.g. Windows then we'll get a compile time failure. 

This could also just be gated under target_os.